lib/ecto/association.ex

```diff
@@ -169,7 +169,7 @@ defmodule Ecto.Association do
   end
 
   def assoc_query(refl, t, query, values) do
-    query = query || %Ecto.Query{from: {"join expression", nil}, prefix: refl.queryable.__schema__(:prefix)}
+    query = query || %Ecto.Query{from: %FromExpr{source: {"join expression", nil}, prefix: refl.queryable.__schema__(:prefix)}}
```

Comment:
We need to discuss this. Should we set the prefix in the whole query or only in from? Maybe we should only have the query prefix and the prefix for every join but not a prefix for the `from`?

Reply:
Actually, I think we should completely ignore the prefix here. Instead, we should change `merge_from` to called as `merge_from(from, assoc)` and then copy both `prefix` and `source` from assoc into from.